pkg/transaction/snapper.go

```diff
@@ -44,45 +40,67 @@ const (
 	maxSnapshots     = 8
 )
 
-type snapperT struct {
+type snapperContext struct {
 	ctx          context.Context
 	s            *sys.System
+	partitions   deployment.Partitions
+	cleanStack   *cleanstack.CleanStack
 	snap         *snapper.Snapper
+	maxSnapshots int
+}
+
+// checkCancelled just checks if the current context is cancelled or not. Returns the context error if cancelled.
+func (sc snapperContext) checkCancelled(err error) error {
```

Comment:
Is this needed? Can this be 

```
if err2 := ctx.Err(); err2 != nil {
    err = err2
}
```

In the deferred calls? (I have not tried this myself yet, channels are always a bit of magic)